src/TaskCreationSection/TaskCreationSection.tsx

```diff
@@ -0,0 +1,132 @@
+/* eslint-disable @typescript-eslint/no-non-null-assertion */
+import { useState, useRef, MutableRefObject } from 'react'
+import { v4 as uuid } from 'uuid'
+import AddTaskButton from './CategoryCreationSection/AddTaskButton'
+import CategoryPicker from './CategoryPicker/CategoryPicker'
+import DatePicker from './DatePicker'
+import TaskImportance from './TaskImportance'
+import Alert, {
+    AlertVariant,
+    showAlert,
+    useAlertState,
+} from '../Components/Alert'
+import { useThemeContext } from '../Contexts/ThemeContext'
+import {
+    CategoryContextProvider,
+    CategoryType,
+} from '../Contexts/CategoryContext'
+import { useTaskContext, defaultTask } from '../Contexts/TaskContext'
+import TaskInput from './CategoryCreationSection/TaskInput'
+import scrollToBottom from '../Helpers/scrollToBottom'
+
+type InputRefType = MutableRefObject<HTMLInputElement | null>
+
+const TaskCreationSection = () => {
+    const { colorStyle, setColorStyle } = useThemeContext()
+    const [newTask, setNewTask] = useState({
+        ...defaultTask,
+    })
+    const { taskList, setTaskList, categoryTabs } = useTaskContext()
+    const inputRef: InputRefType = useRef<HTMLInputElement>(null)
+    const [isSelectDateChecked, setIsSelectDateChecked] = useState(false)
+    const [isCorrectTyped, setIsCorrectTyped] = useState(true)
```

Comment:
maybe we don't need the state to hold this information, let's discuss on the call